channels/worker.py

```diff
@@ -85,11 +93,11 @@ def run(self):
         while not self.termed:
             self.in_job = False
             channel, content = self.channel_layer.receive_many(channels, block=True)
-            self.in_job = True
             # If no message, stall a little to avoid busy-looping then continue
             if channel is None:
                 time.sleep(0.01)
                 continue
+            self.in_job = True
```

Comment:
I moved setting the flag here because getting `(channel, content)` doesn't always means that the worker get a message and have something to process, I encountered it when covering a multiple-workers case with tests, let me know if you disagree and I'm missing something here.